WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -0,0 +1,114 @@
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedChargeReader(let retrySearch, let endSearch):
+            return .alert(.connectingFailedChargeReader(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedChargeReaderAlertViewModel(
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedUpdateAddress(let wcSettingsAdminURL, let retrySearch, let endSearch):
+            return .alert(.connectingFailedUpdateAddress(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel(
+                    settingsAdminUrl: wcSettingsAdminURL,
+                    retrySearchAction: retrySearch,
+                    cancelSearchAction: endSearch)))
+
+        case .foundReader(let name, let connect, let continueSearch, let endSearch):
+            return .alert(.foundReader(
+                viewModel: PointOfSaleCardPresentPaymentFoundReaderAlertViewModel(
+                    readerName: name,
+                    connectAction: connect,
+                    continueSearchAction: continueSearch,
+                    endSearchAction: endSearch)))
+
+        case .updateProgress(let requiredUpdate, let progress, let cancelUpdate):
+            return .alert(.updatingReader(viewModel: CardPresentPaymentUpdatingReaderAlertViewModel()))
+
+        case .updateFailed(let tryAgain, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedNonRetryable(let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedLowBattery(let batteryLevel, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        /// Payment messages
+        case .preparingForPayment(cancelPayment: let cancelPayment):
+            return .message(.preparingForPayment)
+
+        case .tapSwipeOrInsertCard(inputMethods: let inputMethods, cancelPayment: let cancelPayment):
+            return .message(.tapSwipeOrInsertCard)
+
+        case .success(done: let done):
+            return .message(.success)
+
+        case .error(error: let error, tryAgain: let tryAgain, cancelPayment: let cancelPayment):
+            return .message(.error)
+
+        case .errorNonRetryable(error: let error, cancelPayment: let cancelPayment):
+            return .message(.nonRetryableError)
+
+        case .processing:
+            return .message(.processing)
+
+        case .displayReaderMessage(message: let message):
+            return .message(.displayReaderMessage(message: message))
+
+        case .cancelledOnReader:
+            return .message(.cancelledOnReader)
+
+        /// Not-yet supported types
```

Comment:
just a clarifying question, do we plan to support these in M1?

Reply:
We might need to do `validatingOrder` – it's the first step after the reader connects though, so it's probably a simple implementation where we do exactly the same as `preparingForPayment`.

`selectSearchType` is about choosing between Tap to Pay and a Bluetooth reader, during the connection flow, so we don't need to do that in the MVP at all. We don't actually use it in the main app either, it's leftover from an old design.